core/include/scattered_array.h

```diff
@@ -0,0 +1,127 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2018, Linaro Limited
+ */
+#ifndef __SCATTERED_ARRAY_H
+#define __SCATTERED_ARRAY_H
+
+#include <compiler.h>
+#include <keep.h>
+
+/*
+ * A scattered array is assembled from items declared in different source
+ * files whithout the need for any further linker support.
```

Comment:
Minor: what do you mean by "[without] the need for further linker support"? We do rely on the linker to assemble the array elements in the correct order, and that is specified in the linker script, so I'd rather emphasize that in this comment.

Reply:
I'll update the comment.